minimal_data_sharing/minimal_lock_free_queue_logging.cpp

```diff
@@ -0,0 +1,100 @@
+// Copyright 2022 Carlos San Vicente
+// Copyright 2016 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <chrono>
+#include <memory>
+#include <string>
+#include <thread>
+
+#include "rclcpp/rclcpp.hpp"
+#include "std_msgs/msg/string.hpp"
+
+#include "iceoryx_utils/concurrent/lockfree_queue.hpp"
+#include "string/string.hpp"
+
+using namespace std::chrono_literals;
+
+class MinimalSubscriber : public rclcpp::Node
+{
+public:
+  using LogMessage = apex::string256_t;
+  using LogMessageQueue = iox::concurrent::LockFreeQueue<LogMessage, 100>;
+
+  MinimalSubscriber()
+  : Node("minimal_subscriber"), count_(0)
+  {
+    publisher_ = this->create_publisher<std_msgs::msg::String>("topic", 10);
+    auto timer_callback =
+      [this]() -> void {
+        auto message = std_msgs::msg::String();
+        message.data = "Hello, world! " + std::to_string(this->count_++);
+        logger_queue_.push(apex::varargs_to_string("Publishing: ", message.data.c_str()));
+        this->publisher_->publish(message);
+      };
+    timer_ = this->create_wall_timer(500ms, timer_callback);
+
+    subscription_ = this->create_subscription<std_msgs::msg::String>(
+      "topic",
+      10,
+      [this](std_msgs::msg::String::UniquePtr msg) {
+        logger_queue_.push(apex::varargs_to_string("I heard: ", msg->data.c_str()));
+      });
+
+    // Note this is not an example of how to implement a real-time logger
+    // This only shows how pass data from a real-time thread to another
+    // thread to handle the data in a non real-time context
```

Comment:
You say it's not an example of how to implement a RT logger.... but it seems to work (if the magical values are chosen correctly...) :smile: :grimacing: 

Not for this demo, and just a note: I have been [playing with lttng](https://github.com/shuhaowu/rt-demo/tree/master/examples/lttng_ust_example) and it has a way to emit logs via the [`tracef` function](https://lttng.org/man/3/lttng_ust_tracef/v2.13/). I haven't played with it enough to know how good the ergonomics are, but it can be interesting.